test/SlowTests/Server/Documents/DataArchival/DataArchivalIndexingTests.cs

```diff
@@ -0,0 +1,483 @@
+            var retires = SystemTime.UtcNow.AddMinutes(5);
+            using (var session = store.OpenAsyncSession())
+            {
+                await session.StoreAsync(company);
+                var metadata = session.Advanced.GetMetadataFor(company);
+                metadata[Constants.Documents.Metadata.ArchiveAt] = retires.ToString(DefaultFormat.DateTimeOffsetFormatsToWrite);
+                await session.SaveChangesAsync();
+            }
+
+            // Make sure that the company is not skipped while indexing yet
+            using (var session = store.OpenAsyncSession())
+            {
+                await Indexes.WaitForIndexingAsync(store);
+                var companies = await session.Query<Companies_NamesCount.Result, Companies_NamesCount>().ToListAsync();
+                Assert.Equal(1, companies.Count);
+            }
+
+            // Activate the archival
+            await SetupDataArchival(store);
+
+            var database = await Databases.GetDocumentDatabaseInstanceFor(store);
+            database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
+            var documentsArchiver = database.DataArchivist;
+            await documentsArchiver.ArchiveDocs();
+
+            using (var session = store.OpenAsyncSession())
+            {
+                var archivedCompany = await session.LoadAsync<Company>(company.Id);
+                var metadata = session.Advanced.GetMetadataFor(archivedCompany);
+                Assert.DoesNotContain(Constants.Documents.Metadata.ArchiveAt, metadata.Keys);
+                Assert.Contains(Constants.Documents.Metadata.Collection, metadata.Keys);
+                Assert.Contains(Constants.Documents.Metadata.Archived, metadata.Keys);
+                Assert.Equal(true, metadata[Constants.Documents.Metadata.Archived]);
+                
+                // Make sure that the company is skipped while indexing (auto map index)
+                var companies = await session.Query<Companies_NamesCount.Result, Companies_NamesCount>().ToListAsync();
+                await AssertWaitForCountAsync(() => Task.FromResult(companies) , 0);
+            }
+        }
+    }
+
+    public class Companies_AddressText: AbstractIndexCreationTask<Company, Companies_AddressText.IndexEntry>
+    {
+        public class IndexEntry
+        {
+            public string AddressText{ get; set; }
+        }
+       
+        public Companies_AddressText()
+        {
+            Map = companies => from company in companies 
+                select new IndexEntry
+                {
+                    AddressText = company.Address1
+                };
+            
+            Index(x => x.AddressText, FieldIndexing.Search);
+        }
+    }
+     
+    [RavenTheory(RavenTestCategory.Indexes)]
+    [RavenData(SearchEngineMode = RavenSearchEngineMode.All)]
+    public async Task CanArchiveEntity_WillNotBeRelevantInMapSearchIndex(Options options)
```

Comment:
Can we express explicitly what is gonna happen with an archived document instead of using "relevant" word? I think the purpose of this test is to verify that a document which gets archived will be skipped / deleted from an index, right?

So we could rename the test to `WillRemoveDocumentOnceItGetsArchived` or  something like that. I'd like to have the names of test to be as much meaningful and descriptive as possible so one can easily figure out what is it about without the need to read the test code. Please go over all tests you added so far and see if that can be improved.

Reply:
Done, mentioned here https://github.com/ravendb/ravendb/pull/16732#discussion_r1287237047